src/main/java/com/github/exadmin/sourcesscanner/exclude/ExcludeSignatureItem.java

```diff
@@ -32,6 +32,9 @@ public void setFileHash(String fileHash) {
         this.fileHash = fileHash;
     }
 
+    /**
+    * Override equals method for the ExcludeSegnatureItem-class
+    **/
     @Override
     public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
```

Comment:
Use instanceof pattern matching for cleaner null and type checking.

```suggestion
        if (!(o instanceof ExcludeSignatureItem other)) return false;
```

#deepseek-review:inline